websites/T/TuneIn/presence.ts

```diff
@@ -1,85 +1,119 @@
-					),
-					presence.timestampFromFormat(
-						document.querySelector("#scrubberDuration").textContent
-					)
-				),
-				paused = document
-					.querySelector("#innerAppContent")
-					.querySelectorAll('[data-testid="player-status-paused"]');
 
-			presenceData.details = title;
-			if (title.length > 128)
-				presenceData.details = `${title.substring(0, 125)}...`;
+			if (conditions.isLive.textContent === "") {
+				if (conditions.isStopped) {
+					delete presenceData.startTimestamp;
+					presenceData.smallImageKey = Assets.Pause;
+					presenceData.smallImageText = strings.pause;
+				} else {
+					if (!settings.private && settings.timestamps) {
+						const timeElapsed = presence.timestampFromFormat(
+							document
+								.querySelector("#scrubberElapsed")
+								.getAttribute("data-elapsedtime")
+						);
+						presenceData.startTimestamp =
+							Math.floor(Date.now() / 1000) - timeElapsed;
+					}
+					presenceData.smallImageKey = Assets.Live;
+					presenceData.smallImageText = strings.live;
+				}
+			} else {
+				if (!settings.private && settings.timestamps) {
+					const elapsed =
+							document.querySelector("#scrubberElapsed").textContent,
+						duration = document.querySelector("#scrubberDuration").textContent;
+
+					if (elapsed !== "00:00" || duration !== "") {
+						const timestamps = presence.getTimestamps(
+							presence.timestampFromFormat(elapsed),
+							presence.timestampFromFormat(duration)
+						);
 
-			presenceData.state = author;
-			if (author.length > 128)
-				presenceData.state = `${author.substring(0, 125)}...`;
+						presenceData.startTimestamp = timestamps[0];
+						presenceData.endTimestamp = timestamps[1];
+					}
+				} else {
+					delete presenceData.startTimestamp;
+					delete presenceData.endTimestamp;
+				}
 
-			if (paused[0]) {
-				delete presenceData.startTimestamp;
-				delete presenceData.endTimestamp;
-				presenceData.smallImageKey = Assets.Pause;
-				presenceData.smallImageText = (await strings).pause;
-			} else {
-				presenceData.smallImageKey = Assets.Play;
-				presenceData.smallImageText = (await strings).play;
+				if (conditions.isPaused) {
+					delete presenceData.startTimestamp;
+					delete presenceData.endTimestamp;
+					presenceData.smallImageKey = Assets.Pause;
+					presenceData.smallImageText = strings.pause;
+				} else if (conditions.isStopped) {
+					delete presenceData.startTimestamp;
+					delete presenceData.endTimestamp;
+					presenceData.smallImageKey = Assets.Stop;
+					presenceData.smallImageText = strings.pause;
+				} else {
+					presenceData.smallImageKey = Assets.Play;
+					presenceData.smallImageText = strings.play;
+				}
 			}
-			presenceData.endTimestamp = timestamps.pop();
-		}
-		if (!presenceData.details) presence.setActivity();
-		else presence.setActivity(presenceData);
-	} else presence.setActivity();
+		} else return presence.setActivity(presenceData);
```

Comment:
This else statement may be redundant

Reply:
> This else statement may be redundant

In fact, it ended up being redundant, I've fixed it! 👍